src/Core/Binding/IBindingConfigProvider.cs

```diff
@@ -44,11 +44,9 @@ public interface IBindingConfigProvider
     /// For C++ it will be in a json file in a Sonar-specific format</remarks>
     public interface IBindingConfig
     {
-        string FilePath { get; }
-
         /// <summary>
         /// Saves the file, replacing any existing file
         /// </summary>
-        void Save();
+        void Save(ISourceControlledFileSystem sourceControlledFileSystem);
```

Comment:
I was debating if I should pass this as ctor parameter, and I think I should. It would require to pass it down from `BindingController`, but I think it would make for a cleaner API.

Reply:
I'm not convinced. Let's discuss it offline.